src/OrleansAzureUtils/MultiClusterNetwork/GossipTableInstanceManager.cs

```diff
@@ -0,0 +1,327 @@
+        }
+
+        internal async Task<bool> TryCreateGatewayEntryAsync(GatewayEntry entry)
+        {
+            var row = new GossipTableEntry()
+            {
+                PartitionKey = GlobalServiceId,
+                RowKey = GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId),
+                Status = entry.Status.ToString(),
+                GossipTimestamp = entry.HeartbeatTimestamp
+            };
+
+            return (await TryCreateTableEntryAsync(row).ConfigureAwait(false) != null);
+        }
+
+
+        internal async Task<bool> TryUpdateGatewayEntryAsync(GatewayEntry entry, GossipTableEntry row, string eTag)
+        {
+            //Debug.Assert(row.ETag == eTag);
+            //Debug.Assert(row.PartitionKey == GlobalServiceId);
+            //Debug.Assert(row.RowKey == GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId));
+            
+            row.Status = entry.Status.ToString();
+            row.GossipTimestamp = entry.HeartbeatTimestamp;
+
+            return (await TryUpdateTableEntryAsync(row, eTag).ConfigureAwait(false) != null);
+        }
+
+        internal Task<bool> TryDeleteGatewayEntryAsync(GossipTableEntry row, string eTag)
+        {
+            return TryDeleteTableEntryAsync(row, eTag);
+        }
+
+        internal async Task<int> DeleteTableEntries()
+        {
+            var entries = await storage.ReadAllTableEntriesForPartitionAsync(GlobalServiceId).ConfigureAwait(false);
+            var entriesList = new List<Tuple<GossipTableEntry, string>>(entries);
+            if (entriesList.Count <= AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS)
+            {
+                await storage.DeleteTableEntriesAsync(entriesList).ConfigureAwait(false);
+            }
+            else
+            {
+                List<Task> tasks = new List<Task>();
+                foreach (var batch in entriesList.BatchIEnumerable(AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS))
+                {
+                    tasks.Add(storage.DeleteTableEntriesAsync(batch));
+                }
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+            return entriesList.Count;
+        }
+
+
+        /// <summary>
+        /// Try once to conditionally update a data entry in the Azure table. Returns null if etag does not match.
+        /// </summary>
+        private async Task<string> TryUpdateTableEntryAsync(GossipTableEntry data, string dataEtag, [CallerMemberName]string operation = null)
+        {
+            try
+            {
+                return await storage.UpdateTableEntryAsync(data, dataEtag).ConfigureAwait(false);
+            }
+            catch (Exception exc)
+            {
+                HttpStatusCode httpStatusCode;
+                string restStatus;
+                if (!AzureStorageUtils.EvaluateException(exc, out httpStatusCode, out restStatus)) throw;
```

Comment:
well, the reality is that this code doesn't rely on the scheduling context nor the logical call context at all, so it doesn't need to execute in it just to perform some minor continuations.
As long as the callers of these methods (ie: system targets, grains) await them without using `.ConfigureAwait(false)`, then THEY will execute continuations from the completed tasks on the same scheduling context, which is what we need.